gossip/components/processor.c

```diff
@@ -198,35 +196,47 @@ static retcode_t process_packet(processor_t const *const processor,
   retcode_t ret = RC_OK;
   neighbor_t *neighbor = NULL;
   flex_trit_t hash[FLEX_TRIT_SIZE_243];
+  char *protocol = NULL;
 
   if (processor == NULL || packet == NULL) {
     return RC_NULL_PARAM;
   }
 
+  rw_lock_handle_rdlock(&processor->node->neighbors_lock);
+
   neighbor =
-      neighbor_find_by_endpoint(processor->node->neighbors, &packet->source);
+      neighbors_find_by_endpoint(processor->node->neighbors, &packet->source);
+  protocol = packet->source.protocol == PROTOCOL_TCP ? "tcp" : "udp";
 
   if (neighbor) {
+    log_debug(PROCESSOR_LOGGER_ID,
+              "Processing packet from tethered node %s://%s:%d\n", protocol,
+              neighbor->endpoint.host, neighbor->endpoint.port);
     neighbor->nbr_all_tx++;
 
     log_debug(PROCESSOR_LOGGER_ID, "Processing transaction bytes\n");
     if ((ret = process_transaction_bytes(processor, neighbor, packet, hash)) !=
         RC_OK) {
       log_warning(PROCESSOR_LOGGER_ID, "Processing transaction bytes failed\n");
-      return ret;
+      goto done;
     }
 
     log_debug(PROCESSOR_LOGGER_ID, "Processing request bytes\n");
     if ((ret = process_request_bytes(processor, neighbor, packet, hash)) !=
         RC_OK) {
       log_warning(PROCESSOR_LOGGER_ID, "Processing request bytes failed\n");
-      return ret;
+      goto done;
     }
   } else {
+    log_debug(PROCESSOR_LOGGER_ID,
+              "Discarding packet from non-tethered node %s://%s:%d\n", protocol,
```

Comment:
This should have some feedback via return codes to close the connection?

Reply:
TCP connections from non-tethered nodes are automatically closed from within the `tcp_receiver` so this is mostly for incoming UDP packet from non-tethered nodes.
Though, it can happen that a previously paired TCP neighbor is unpaired with the API call without the connection being closed, but returning from `process_packet` won't allow the receiver to close the connection since it doesn't have access to it.